modules/bioinspired/src/retina.cpp

```diff
@@ -649,7 +663,7 @@ void RetinaImpl::_convertValarrayBuffer2cvMat(const std::valarray<float> &grayMa
             for (unsigned int j=0;j<nbColumns;++j)
             {
                 cv::Point2d pixel(j,i);
-                outMat.at<unsigned char>(pixel)=(unsigned char)*(valarrayPTR++);
+                outMat.at<unsigned char>(pixel)=(unsigned char)round(*(valarrayPTR++));
```

Comment:
What kind of problem is fixed here (and below)?

BTW, it is better to use `cvRound()` instead. Also saturation may be added (if necessary).